tensorflow/core/framework/types.h

```diff
@@ -68,9 +68,17 @@ class DeviceType {
 std::ostream& operator<<(std::ostream& os, const DeviceType& d);
 
 // Convenient constants that can be passed to a DeviceType constructor
-extern const char* const DEVICE_CPU;   // "CPU"
-extern const char* const DEVICE_GPU;   // "GPU"
-extern const char* const DEVICE_SYCL;  // "SYCL"
+#if !defined(TF_EXTERN)
```

Comment:
Maybe this is a better place: https://www.github.com/tensorflow/tensorflow/blob/master/tensorflow/core/platform/macros.h#L42

Perhaps `TF_DLLEXPORT` is a more standard name:
```
#ifdef TF_COMPILE_LIBRARY
# define TF_DLLEXPORT __declspec(dllimport)
#else
# define TF_DLLEXPORT __declspec(dllexport)
#endif   // TF_COMPILE_LIBRARY
```

In time, we'll export the symbols (whitelist) rather than blacklist.

CC: @skye